aspnetcore/signalr/publish-to-azure-web-app.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Publish an ASP.NET Core SignalR app to Azure Web App
+author: rachelappel 
+description: Publish an ASP.NET Core SignalR app to Azure Web App
+manager: wpickett
+ms.author: rachelap
+ms.custom: mvc
+ms.date: 04/20/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/publish-to-azure-web-app
+---
+
+# Publish an ASP.NET Core SignalR app to an Azure Web App
+
+[Azure Web App](https://docs.microsoft.com/azure/app-service/app-service-web-overview) is a [Microsoft cloud computing](https://azure.microsoft.com/) platform service for hosting web apps, including ASP.NET Core.
+
+## Publish the app
+
+Visual Studio provides built-in tools for publishing to an Azure Web App. Visual Studio Code user can use [Azure CLI](https://docs.microsoft.com/cli/azure) commands to publish apps to Azure. This article covers publishing using  the tools in Visual Studio. To publish an app using Azure CLI, see [Publish an ASP.NET Core app to Azure with command line tools](xref:tutorials/publish-to-azure-webapp-using-cli).
+
+Right-click on the project in **Solution Explorer** and select **Publish**. Confirm that **Create new** is checked in the **Pick a publish target** dialog, and select **Publish**.
+
+![Pick publish target](publish-to-azure-web-app/_static/pick-publish-target-dialog.png)
+
+Enter the following information in the **Create App Service** dialog and select **Create**.
+
+| Item | Description |
+| ---- | ----------- |
+| **App name** | A unique name of the app. |
+| **Subscription** | The Azure subscription that the app uses. |
+| **Resource Group** | The group of related resources to which the app belongs.  |
+| **Hosting Plan** | The pricing plan for the web app. |
+
+![Create app service](publish-to-azure-web-app/_static/create-app-service-dialog.png)
+
+Visual Studio completes the following tasks:
+
+* Creates a Publish Profile containing publish settings.
+* Creates or uses an existing *Azure Web App* with the provided details.
+* Publishes the app.
+* Launches a browser, with the published web app loaded.
+
+Notice the format of the URL for the app is *{app name}.azurewebsites.net*. For example, an app named `SignalRChattR` has a URL that looks like https://signalrchattr.azurewebsites.net.
+
+![Appname-azurewebsites-net](publish-to-azure-web-app/_static/app-azurewebsites-net.png)
+
+If an HTTP 502.2 error occurs, see [Deploy ASP.NET Core preview release to Azure App Service](xref:host-and-deploy/azure-apps/?view=aspnetcore-2.1#deploy-aspnet-core-preview-release-to-azure-app-service) to resolve it.
+
+## Configure SignalR web app
+
+ASP.NET Core SignalR apps that are published as an Azure Web App must have [WebSockets](xref:fundamentals/websockets) and [ARR Affinity](https://en.wikipedia.org/wiki/Application_Request_Routing) enabled. To configure and manage the published app, use the [Azure Portal](https://portal.azure.com).
```

Comment:
> ASP.NET Core SignalR apps that are published as an Azure Web App must have [WebSockets](xref:fundamentals/websockets) and [ARR Affinity](https://en.wikipedia.org/wiki/Application_Request_Routing) enabled.

ARR Affinity is always required, but WebSockets is only required if you want to enable the WebSockets transport (which is recommended). I'd suggest rephrasing to:

ASP.NET Core SignalR apps that are published as an Azure Web App must have ARR Afinity enabled. WebSockets should also be enabled, to allow the WebSockets transport to function. To configure and manage the published app, ...

Reply:
We should also direct users to this page: https://docs.microsoft.com/en-us/azure/azure-subscription-service-limits#app-service-limits

It documents the limits of WebSockets on Azure App Service. Most importantly, the Free tier only allows 5 concurrent WebSocket connections. I believe there are also concurrent connection limits that restrict the other transports on Azure App Service as well, and we should document them as they have been a problem in the past (cc @davidfowl)

Reply:
@anurse Do you want the free tier/5 connections thing specifically pointed in this doc out or just a link saying to be aware of limitations?

Reply:
The latter is probably fine.